src/main/java/duke/query/DeadlineQueryHandler.java

```diff
@@ -0,0 +1,56 @@
+package duke.query;
+
+import duke.DukeException;
+import duke.task.Task;
+import duke.task.TaskTracker;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The DeadlineQueryHandler class handles user queries for adding new deadlines.
+ */
+public class DeadlineQueryHandler extends TaskQueryHandler {
+    protected final String DATETIME_PATTERN = "yyyy-MM-dd HH:mm";
+
+    public DeadlineQueryHandler(TaskTracker tt) {
+        super(tt);
+    }
+
+    /**
+     * Processes a query for adding a new deadline.
+     *
+     * @param query a user input string
+     * @return response from adding deadline
+     * @throws DukeException
+     */
+    @Override
+    public String processQuery(String query) throws DukeException {
+        String[] parsed = QueryParser.parseQuery(query, new String[]{"/by"});
```

Comment:
May I ask why this variable was named `parsed`? Perhaps a name could be chosen to convey the message that is contains the arguments from the parsed command!

Reply:
Yup, this is a good point. I will probably refactor the results of this function into its own class to better convey the intent and context behind the results.